src/async_wrap.cc

```diff
@@ -226,7 +246,27 @@ static PromiseWrap* extractPromiseWrap(Local<Promise> promise) {
   return obj->IsObject() ? Unwrap<PromiseWrap>(obj.As<Object>()) : nullptr;
 }
 
-static void PromiseHook(PromiseHookType type, Local<Promise> promise,
+// Simplified JavaScript hook fast-path for when there is no destroy hook
+static void FastPromiseHook(PromiseHookType type, Local<Promise> promise,
+                        Local<Value> parent) {
+  Local<Context> context = promise->CreationContext();
+
+  Environment* env = Environment::GetCurrent(context);
+  if (env == nullptr) return;
+  TraceEventScope trace_scope(TRACING_CATEGORY_NODE1(environment),
+                              "EnvPromiseHook", env);
+
+  Local<Value> argv[] = {
+    env->isolate_data()->promise_hook_type(static_cast<int>(type)),
```

Comment:
I think it might be quite a bit easier to pass this as an integer to JS, and export `PromiseHookType::kInit`,  `PromiseHookType::kResolve`, etc. to JS as constants? That’s what we usually do in this kind of situation, I think.

Reply:
Good call. I'll see what I can do. 👍

Reply:
Would it be reasonable to just use the existing AsyncHooks constants for `kInit`, `kBefore`, `kAfter` and `kPromiseResolve`? Or should I make a new set? It'd mean those constants have two purposes: tracking handler presence in the fields array _and_ identifying promise hook types. Seems better to me to avoid adding more things when we can, but I could be missing something. 🤔

Reply:
@qard I think that’s reasonable, yes – in that case just be sure to make the translation here in C++ before passing the value to JS so that we do only expose one set of constants to JS